kip-0037.md

```diff
@@ -0,0 +1,147 @@
+---
+KIP: '0037'
+Title: kadena_getAccount_v1 API
+Author: Danillo Felixdaal, Bart Huijgen and Albert Groothedde
+Status: Draft
+Type: Standard
+Category: Interface
+Created: 2025-02-17
+---
+
+# kadena_getAccount_v1 API
+
+## Abstract
+
+This KIP standardizes the `kadena_getAccount_v1` method to provide the active
+account known by the wallet. This allows dApps to present the user with the
+currently selected account and use it for creating and signing transactions.
+
+## Motivation
+
+The `kadena_getAccount_v1` method provides a standardized way to retrieve a
+single, active account from a Kadena wallet. This method is used when the dApp
+does not need to specify which account to retrieve and relies on the wallet's
+current active selection.
+
+## Specification
+
+### Method Definition
+
+**Method Name:** `kadena_getAccount_v1`
+
+#### Parameters
+
+No parameters required. The wallet can return any account that is currently
+active.
+
+### Response
+
+```typescript
+interface AccountInfo {
+  accountName: string; // The unique identifier for the account.
+  networkId: string; // The unique identifier for the network for this account.
+  contract: string; // Identifier for the fungible token contract.
```

Comment:
Should `contract` be optional for accounts that aren't minted yet? Or do wallets always associate accounts with a contract right away?

Also, is it possible for the same `accountName` to be present in multiple contracts? If so, should this field be a list similar to `chainAccounts`?

Reply:
> Also, is it possible for the same accountName to be present in multiple contracts? If so, should this field be a list similar to chainAccounts?

The wallet only returns the ACTIVE account. Since an account is always tied to a contract, this is not a list

Reply:
Added document suggestion related to this here: https://github.com/kadena-io/KIPs/pull/71/files#r2044838690

Resolving this conversation thread